libs/common/src/models/domain/organization.ts

```diff
@@ -196,8 +199,22 @@ export class Organization {
     return this.canManagePolicies;
   }
 
-  get canManageBilling() {
-    return this.isOwner && (this.isProviderUser || !this.hasProvider);
+  get canViewSubscription() {
+    return this.hasProvider && this.providerType === ProviderType.Msp
+      ? this.isProviderUser
+      : this.isOwner;
+  }
+
+  get canEditSubscription() {
+    return this.canViewSubscription && (this.hasProvider ? this.isProviderUser : this.isOwner);
```

Comment:
Then revert this recent change:

```suggestion
    return this.hasProvider ? this.isProviderUser : this.isOwner;
```